cpp/sbml_model_generation/sbml_to_memilio.cpp

```diff
@@ -0,0 +1,1053 @@
+        example << "sim.set_integrator(integrator);" << std::endl;
+        for (size_t i = 0; i < model.getListOfEvents()->size(); i++) {
+            auto event                      = model.getListOfEvents()->get(i);
+            auto trigger                    = event->getTrigger()->getMath();
+            mio::IOResult<std::string> tmax = find_tmax(trigger, model, &lowercase_name);
+            if (!tmax) {
+                mio::log_info("There is an event that I can not implement, I will ignore it.");
+            }
+            else {
+                example << "tmax = " << tmax.value() << ";" << std::endl;
+                example << "sim.advance(tmax);" << std::endl;
+                for (size_t j = 0; j < event->getListOfEventAssignments()->size(); j++) {
+                    auto assignment = event->getListOfEventAssignments()->get(j);
+                    auto variable   = assignment->getVariable();
+                    mio::IOResult<std::string> formatted_variable =
+                        format_event_variable(variable, model, &lowercase_name);
+                    if (!formatted_variable) {
+                        return false;
+                    }
+                    std::string formula = SBML_formulaToL3String(assignment->getMath());
+                    mio::IOResult<std::string> formatted_formula =
+                        format_event_formulas(formula, model, &lowercase_name);
+                    example << formatted_variable.value() << " = " << formatted_formula.value() << ";" << std::endl;
+                }
+            }
+        }
+        example << "sim.advance(tmax + 50); \nauto sir = sim.get_result();" << std::endl;
+
+        example << "sir.print_table({" << table << "});" << std::endl;
+        example << "mio::log_info(\"number total: {}\", sir.get_last_value().sum());" << std::endl;
+
+        example << "const std::string file_name = \"" << lowercase_name << ".dat\";" << std::endl;
+        example << "std::ofstream file(file_name);" << std::endl;
+        example << "if (!file) {  mio::log_error(\"Could not open file for writing: {}\", file_name); return 1; }"
+                << std::endl;
+        example << "mio::log_info(\"Writing output to {}\", file_name);" << std::endl;
+        example << "sir.print_table({" << table << "}, 10, 3, file);" << std::endl;
+        example << "file.close();" << std::endl;
+
+        example << "}";
+        example.close();
+    }
+    else {
+        mio::log_error("Could not open file for writing: example.cpp");
+        return false;
+    }
+
+    return true;
+}
+
+/**
+ * @brief Create the additional lines for the CmakeLists.txt file
+ *
+ * @param filename The name of the file to be processed
+ * @param path The path where the file should be created
+ *
+ * Appends the necessary commands for compilation to the CMakeLists.txt file in path using the filename.
+*/
+bool modify_cmakelists(const std::string& filename, const std::string& path)
```

Comment:
Does this work when run repeatedly on the same file?

Maybe you could do search for the line "add_subdirectory(lowercase_name)" before appending it again?